system-test/speech.js

```diff
@@ -0,0 +1,411 @@
+          assert(apiResponse);
+          assert(apiResponse.results);
+          done();
+        });
+      });
+    });
+    it('recognizes speech from remote GCS audio file', function(done) {
+      speech.recognize({
+        config: {
+          encoding: 'LINEAR16',
+          sampleRate: 16000
+        },
+        audio: {
+          uri: AUDIO_FILES.bridge.gcsUri
+        }
+      }, function(err, response, apiResponse) {
+        assert.ifError(err);
+        assert(Array.isArray(response.results));
+        assert(response.results.length > 0);
+        assert(apiResponse);
+        assert(apiResponse.results);
+        done();
+      });
+    });
+  });
+
+  describe('detect', function() {
+    it('recognizes speech from raw audio', function(done) {
+      speech.detect(AUDIO_FILES.bridge.path, {
+        encoding: 'LINEAR16',
+        sampleRate: 16000
+      }, function(err, response, apiResponse) {
+        assert.ifError(err);
+        assert(Array.isArray(response));
+        assert(response.length > 0);
+        assert(response[0].transcript);
+        assert(response[0].confidence);
+        assert(Array.isArray(apiResponse.results));
+        assert(apiResponse.results.length > 0);
+        done();
+      });
+    });
+    it('recognizes speech from remote GCS audio file', function(done) {
+      speech.detect(AUDIO_FILES.bridge.gcsUri, {
+        encoding: 'LINEAR16',
+        sampleRate: 16000
+      }, function(err, response, apiResponse) {
+        assert.ifError(err);
+        assert(Array.isArray(response));
+        assert(response.length > 0);
+        assert(response[0].transcript);
+        assert(response[0].confidence);
+        assert(Array.isArray(apiResponse.results));
+        assert(apiResponse.results.length > 0);
+        done();
+      });
+    });
+    it('recognizes speech from remote audio file', function(done) {
+      speech.detect(AUDIO_FILES.bridge.httpUri, {
+        encoding: 'LINEAR16',
+        sampleRate: 16000
+      }, function(err, response, apiResponse) {
+        assert.ifError(err);
+        assert(Array.isArray(response));
+        assert(response.length > 0);
+        assert(response[0].transcript);
+        assert(response[0].confidence);
+        assert(Array.isArray(apiResponse.results));
+        assert(apiResponse.results.length > 0);
+        done();
+      });
+    });
+  });
+
+  describe('createRecognizeJob', function() {
+    it('recognizes speech from raw audio', function(done) {
+      fs.readFile(AUDIO_FILES.bridge.path, function(err, audioFile) {
+        assert.ifError(err);
+        speech.createRecognizeJob({
+          config: {
+            encoding: 'LINEAR16',
+            sampleRate: 16000
+          },
+          audio: {
+            content: audioFile
+          }
+        }, function(err, operation, apiResponse) {
+          assert.ifError(err);
+          assert(operation.name);
+          assert(apiResponse.name);
+          setTimeout(function() {
```

Comment:
Are we doing `setTimeout` because we assume the job won't be finished yet? I know @stephenplusplus mentioned earlier that ideally operations mimic their compute counterpart, that being said a `complete` event would be great here.